modules/core/src/ocl.cpp

```diff
@@ -3746,6 +3746,9 @@ int Kernel::set(int i, const KernelArg& arg)
     cl_int status = 0;
     if( arg.m )
     {
+        std::shared_ptr<ocl::OpenCLExecutionContext> pExecCtx = std::static_pointer_cast<ocl::OpenCLExecutionContext>(arg.m->u->allocatorContext);
+        OpenCLExecutionContextScope scope(pExecCtx ? *pExecCtx.get() : ocl::OpenCLExecutionContext::getCurrent());
```

Comment:
> `Kernel::`

> OpenCLExecutionContextScope

We should not change the current context selected by caller.

We need to ensure that:
- kernel is from the current (default) `OpenCLExecutionContext` (or assume that)
- and all passed buffers are compatible with the current (default) `OpenCLExecutionContext` (the same "OpenCL context")

If it is not true, then raise an error.

Reply:
alright

Reply:
How to properly compare to OpenCLExecutionContexts? I'd write the operator, but I don't know what to compare.

Reply:
"OpenCL context" means handle of type `cl_context`.

E.g. compare these values related to `OpenCLExecutionContexts`: `execCtx.getContext().ptr()`

Reply:
Done